server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -434,12 +434,20 @@ def _build_window_subquery() -> Any:
                 *[getattr(agg_table, col.alias) for col in agg_selected],
                 *[getattr(window_table, col[1].alias) for col in window_selected],
             ]
-            merged_query = (
-                self.QUERY_CLS.from_(agg_query)
-                .select(*merged_selected)
-                .inner_join(all_windows_subquery)
-                .on_field(*step.on)
-            )
+            if step.on:
+                merged_query = (
+                    self.QUERY_CLS.from_(agg_query)
+                    .select(*merged_selected)
+                    .inner_join(all_windows_subquery)
+                    .on_field(*step.on)
+                )
+            else:
+                # If there is no `step.on` columns to join, just put the 2 subqueries side by side:
+                merged_query = (
+                    self.QUERY_CLS.from_(agg_query)
+                    .from_(all_windows_subquery)
+                    .select(*merged_selected)
+                )
```

Comment:
Nice :+1: I think you'll have to do a modification like that l.463 as well: In case `keep_original_granularity` is specified, there's also a join on `step.on`

Reply:
done in 4f9b6e1